open-metadata-implementation/adapters/open-connectors/repository-services-connectors/open-metadata-collection-store-connectors/graph-repository-connector/src/main/java/org/odpi/openmetadata/adapters/repositoryservices/graphrepository/repositoryconnector/GraphOMRSMetadataStore.java

```diff
@@ -1146,12 +1147,16 @@ private void updateEntityClassifications(EntitySummary entity, Vertex vertex, Gr
                 log.debug("{} entity remove classification: {}", methodName, existingName);
                 Vertex classificationVertex = existingClassificationVerticesByName.get(existingName);
                 classificationVertex.remove();
-                existingClassificationVerticesByName.remove(existingName);
+                namesToRemoveSet.add(existingName);
```

Comment:
I don't see why this helps. I may be missing something but I can't see what's wrong with removing the unwanted classification as soon as it is identified...

Reply:
@grahamwallis This change fixed a concurrent modification exception, I believe the issue was that we called an iterator .next , then a  .remove then .next on the same map (existingClassificationVerticesByName) or its iterator. The second .next got the exception, presumably because it realises that the collection it is iterating over has changed underneath it. The fix solves the problem as the .remove is not done in the iterator loop. I don't think it applies to the classifier vertices as they are not being iterated over. Have I misunderstood something ?

Reply:
the explanation makes sense to me from the code.

Reply:
Yes - that makes good sense. I hadn't looked closely enough :-) I was wondering whether it was an issue with the removal of the vertices and then the edges (which should be fine). Thanks for the explanation and for fixing!